jdaviz/core/astrowidgets_api.py

```diff
@@ -0,0 +1,521 @@
+
+        use_skycoord : bool
+            If `True`, use ``skycoord_colname`` to mark.
+            Otherwise, use ``x_colname`` and ``y_colname``.
+
+        marker_name : str, optional
+            Name to assign the markers in the table. Providing a name
+            allows markers to be removed by name at a later time.
+
+        Raises
+        ------
+        AttributeError
+            Sky coordinates are given but reference image does not have a valid WCS.
+
+        ValueError
+            Invalid marker name.
+
+        """
+        if marker_name is None:
+            marker_name = self._default_mark_tag_name
+
+        self._validate_marker_name(marker_name)
+        jglue = self.session.application
+
+        # Link markers to reference image data.
+        image = self.state.reference_data
+
+        # TODO: Is Glue smart enough to no-op if link already there?
+        if use_skycoord:
+            if not data_has_valid_wcs(image):
+                raise AttributeError(f'{getattr(image, "label", None)} does not have a valid WCS')
+            sky = table[skycoord_colname]
+            t_glue = Data(marker_name, ra=sky.ra.deg, dec=sky.dec.deg)
+            jglue.data_collection[marker_name] = t_glue
+            jglue.add_link(t_glue, 'ra', image, 'Right Ascension')
+            jglue.add_link(t_glue, 'dec', image, 'Declination')
+        else:
+            t_glue = Data(marker_name, **table[x_colname, y_colname])
+            jglue.data_collection[marker_name] = t_glue
+            jglue.add_link(t_glue, x_colname, image, image.pixel_component_ids[1].label)
+            jglue.add_link(t_glue, y_colname, image, image.pixel_component_ids[0].label)
+
+        try:
+            self.add_data(t_glue)
+        except Exception as e:  # pragma: no cover
+            self.session.hub.broadcast(SnackbarMessage(
+                f"Failed to add markers '{marker_name}': {repr(e)}",
+                color="warning", sender=self))
+        else:
+            # Only can set alpha and color using self.add_data(), so brute force here instead.
+            # https://github.com/glue-viz/glue/issues/2201
+            for key, val in self.marker.items():
+                setattr(jglue.data_collection[jglue.data_collection.labels.index(marker_name)].style, key, val)  # noqa
+
+            self._marktags.add(marker_name)
+
+    def remove_markers(self, marker_name=None):
+        """Remove some but not all of the markers by name used when
+        adding the markers.
+
+        Parameters
+        ----------
+        marker_name : str
+            Name used when the markers were added.
+            If not given, will delete markers added under default name.
+
+        """
+        if marker_name is None:
+            marker_name = self._default_mark_tag_name
+
+        # TODO: How to test manually created tiled viewers in CI?
+        if marker_name not in self._marktags:  # pragma: no cover

```

Comment:
In Glue-land, there is nothing stopping one viewer to remove markers added by another viewer because data are global, but Astrowidgets internally tracks the marker names and one viewer cannot see what is being tracked by another viewer, and I don't feel like polling all the viewers that can be opened in Imviz for this operation, so for now, it is simpler to disallow one viewer from removing markers added by another viewer.

Reply:
Interesting point -- I hadn't even considered the possibility that there would be One Table to Rule Them All...

Reply:
I think this is very specific to Glue. I don't think markers are global in DS9, are they?

Reply:
I'm not sure, I haven't used it much...

Reply:
Pretty sure I had to give `tvmark` a frame number back in the days, but it was so long ago...

Reply:
I'm pretty sure (based on how tk works) that they are global but ds9 hides their global-ness using some obtuse constructs. 

At any rate, see my thoughts further up about this.  I think this is fine/good as a starting point, but I think we might want to have the imviz helper class here provide a way for the user to quickly get what they want.  That way the viewers themselves can maintain their separation but the user who thinks of the data together (i.e. the way glue wants them to) can access things that are aware of that at the jdaviz level.